File: backend/src/main/java/net/filmscout/backend/services/abstracts/TheMovieDbService.java
package net.filmscout.backend.services.abstracts;

import net.filmscout.backend.models.Movie;
import net.filmscout.backend.models.TvSerie;
import net.filmscout.backend.models.enums.TimeWindow;

import java.util.List;

public interface TheMovieDbService {
    List<Movie> getTrendingMovies(TimeWindow timeWindow, Integer limit);
    List<Movie> getPopularMovies(Integer limit);
    List<Movie> getUpcomingMovies(Integer limit);
    List<TvSerie> getTrendingTvSeries(TimeWindow timeWindow, Integer limit);
    List<TvSerie> getPopularTvSeries(Integer limit);
    List<TvSerie> getOnTheAirTvSeries(Integer limit);
    List<TvSerie> getAiringTodayTvSeries(Integer limit);
}

File: frontend/src/services/TheMovieDbService.ts
import axios from 'axios';

export default class TheMovieDbService {
    getTrendingMovies(timeWindow: string, limit = 8) {
        return axios.get(`/api/themoviedb/trending-movies/${timeWindow}`, {
            params: {
                limit: limit
            }
        });
    }

    getPopularMovies(limit = 8) {
        return axios.get('/api/themoviedb/popular-movies', {
            params: {
                limit: limit
            }
        });
    }

    getUpcomingMovies(limit = 8) {
        return axios.get('/api/themoviedb/upcoming-movies', {
            params: {
                limit: limit
            }
        });
    }

    getTrendingTvSeries(timeWindow: string, limit = 8) {
        return axios.get(`/api/themoviedb/trending-tv-series/${timeWindow}`, {
            params: {
                limit: limit
            }
        });
    }

    getPopularTvSeries(limit = 8) {
        return axios.get('/api/themoviedb/popular-tv-series', {
            params: {
                limit: limit
            }
        });
    }

    getOnTheAirTvSeries(limit = 8) {
        return axios.get('/api/themoviedb/on-the-air-tv-series', {
            params: {
                limit: limit
            }
        });
    }
}